phantomjs-env/src/test/scala/org/scalajs/jsenv/phantomjs/PhantomJSSuite.scala

```diff
@@ -4,5 +4,49 @@ import org.scalajs.jsenv.test._
 
 import org.junit.runner.RunWith
 
-@RunWith(classOf[JSEnvSuiteRunner])
-class PhantomJSSuite extends JSEnvSuite(JSEnvSuiteConfig(new PhantomJSEnv))
+import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
+
+import org.junit.runner.Runner
+import org.junit.runners.Suite
+import org.junit.runners.parameterized.{TestWithParameters, BlockJUnit4ClassRunnerWithParameters}
+import org.junit.runners.model.TestClass
+import org.junit.runner.manipulation.Filter
+import org.junit.runner.Description
+
+@RunWith(classOf[PhantomJSSuiteRunner])
+class PhantomJSSuite private (private[phantomjs] val _config: JSEnvSuiteConfig)
+    extends JSEnvSuite(_config) {
+
+  def this() = this(JSEnvSuiteConfig(new PhantomJSEnv))
+}
+
+/** A runner that wraps `JSEnvSuiteRunner` and ignores `largeMessageTest`.
+ *
+ *  This works around [[https://github.com/scala-js/scala-js/issues/3476]]:
+ *  `largeMessageTest` sends messages that are not valid UTF-16 strings, which
+ *  is invalid.
+ */
+final class PhantomJSSuiteRunner(root: Class[_], base: JSEnvSuiteRunner)
+    extends Suite(root, List[Runner](base).asJava) {
+
+  def this(root: Class[_], config: JSEnvSuiteConfig) =
+    this(root, new JSEnvSuiteRunner(config))
+
+  /** Constructor for reflective instantiation via `@RunWith`. */
+  def this(suite: Class[_ <: PhantomJSSuite]) =
+    this(suite, suite.newInstance()._config)
+
+  /** Constructor for instantiation in a user defined Runner. */
+  def this(config: JSEnvSuiteConfig) = this(null, config)
+
+  // Apply a filter that ignores all tests whose method name is `largeMessageTest`
+  base.filter(new Filter {
+    def describe(): String = "Ignore largeMessageTest"
+
+    def shouldRun(description: Description): Boolean = {
+      description.getMethodName == null ||
+      !description.getMethodName.startsWith("largeMessageTest")
```

Comment:
Reference scala-js/scala-js#3476?

Reply:
It's referenced in the runner's Scaladoc. I assume that was enough, since the whole purpose of the runner is to address that issue.